### ISSUE TITLE
Concurrent thredds requests are not synchronized against S3

### ISSUE THREAD

**Issue by danfruehauf** [aodn/aodn-portal#2149]
This is related to usage of the abomination controller (which uses thredds).

## Steps to reproduce

 * Add an abomination controller collection (SRS_SST, SRS_OC, ACORN, etc)

## What should happen

 * Abomination controller accesses thredds.aodn.org.au WMS endpoint
 * File is being downloaded and cached just once by thredds

## What does happen

 * Abomination controller accesses thredds.aodn.org.au WMS endpoint
 * File is cached multiple times from S3

**Comment by danfruehauf** [comment]
Should also be communicated to upstream if there is a fix.

Probably the simplest fix for the time being is to lock when trying to download the same S3 URI. See:
https://github.com/aodn/thredds/blob/s3_datasets_stable/cdm/src/main/java/thredds/crawlabledataset/CrawlableDatasetAmazonS3.java#L284

**Comment by danfruehauf** [comment]
This also results in more disk usage, due to the multiple downloads of the same object.

**Comment by bhasin85** [comment]
Verified that the duplicate copies of the cached files are not getting generated.

**Comment by danfruehauf** [comment]
@bhasin85 I'm afraid this does happen (I wrote the code, I know it does). More than this is a bug - it is a shortcoming of the originally written code which I overlooked. You guys may choose to fix it or not, but that problem still exists.

If you couldn't recreate it, I suggest one of:
 * Auditing the code and programmatically understanding the problem
 * Trying to reproduce it with reasonably large files of SRS or GSLA (40MB+ files). It's less likely to happen with ACORN (200KB files)

Good luck!
